File: android/app/src/main/java/com/apiscan/app/data/api/HornetDtos.kt
package com.apiscan.app.data.api

import com.google.gson.annotations.SerializedName

// MARK: - Stats

data class HornetStats(
    @SerializedName("total_caught")       val totalCaught: Int,
    @SerializedName("total_nests")        val totalNests: Int,
    @SerializedName("destroyed_nests")    val destroyedNests: Int,
    @SerializedName("pending_sightings")  val pendingSightings: Int,
    @SerializedName("confirmed_sightings") val confirmedSightings: Int,
    @SerializedName("total_traps")        val totalTraps: Int = 0
)

// MARK: - Catches

data class HornetCatchCreate(
    val latitude: Double?,
    val longitude: Double?,
    val count: Int,
    @SerializedName("reporter_name") val reporterName: String?
)

data class HornetCatchOut(
    val id: String,
    val count: Int,
    val latitude: Double?,
    val longitude: Double?,
    @SerializedName("created_at") val createdAt: String
)

// MARK: - Nests

data class HornetNestCreate(
    val latitude: Double,
    val longitude: Double,
    @SerializedName("reporter_name") val reporterName: String?,
    val notes: String?,
    @SerializedName("photo_url") val photoUrl: String?
)

data class HornetNestOut(
    val id: String,
    val latitude: Double,
    val longitude: Double,
    val status: String,
    @SerializedName("reporter_name") val reporterName: String?,
    val notes: String?,
    @SerializedName("photo_url") val photoUrl: String?,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String
)

data class HornetNestGeoJSON(
    val type: String,
    val features: List<HornetNestFeature>
)

data class HornetNestFeature(
    val type: String,
    val geometry: HornetNestGeometry,
    val properties: HornetNestProperties
)

data class HornetNestGeometry(
    val type: String,
    val coordinates: List<Double>   // [longitude, latitude]
)

data class HornetNestProperties(
    val id: String,
    val status: String,
    @SerializedName("reporter_name") val reporterName: String?,
    val notes: String?,
    @SerializedName("photo_url") val photoUrl: String?,
    @SerializedName("created_at") val createdAt: String
)

// MARK: - Sightings

data class HornetSightingCreate(
    @SerializedName("photo_url") val photoUrl: String,
    val description: String?,
    @SerializedName("reporter_name") val reporterName: String?,
    val latitude: Double?,
    val longitude: Double?
)

data class HornetSightingOut(
    val id: String,
    @SerializedName("photo_url") val photoUrl: String,
    val description: String?,
    @SerializedName("reporter_name") val reporterName: String?,
    val status: String,
    @SerializedName("yes_votes") val yesVotes: Int,
    @SerializedName("no_votes") val noVotes: Int,
    @SerializedName("created_at") val createdAt: String,
    val latitude: Double?,
    val longitude: Double?
)

data class HornetVoteRequest(val vote: String)

// MARK: - Traps

data class HornetTrapCreate(
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val notes: String?,
    @SerializedName("owner_name") val ownerName: String?
)

data class HornetTrapCatchCreate(
    val count: Int,
    @SerializedName("caught_on") val caughtOn: String   // YYYY-MM-DD
)

data class HornetTrapCatchOut(
    val id: String,
    @SerializedName("trap_id") val trapId: String,
    val count: Int,
    @SerializedName("caught_on") val caughtOn: String,
    @SerializedName("created_at") val createdAt: String
)

data class HornetTrapOut(
    val id: String,
    @SerializedName("access_code") val accessCode: String,
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val notes: String?,
    @SerializedName("owner_name") val ownerName: String?,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("total_caught") val totalCaught: Int,
    val catches: List<HornetTrapCatchOut>
)

data class HornetTrapNearbyOut(
    @SerializedName("access_code") val accessCode: String,
    val name: String,
    val latitude: Double,
    val longitude: Double,
    @SerializedName("distance_m") val distanceM: Int,
    @SerializedName("total_caught") val totalCaught: Int
)

data class HornetTrapsGeoJSON(
    val type: String,
    val features: List<HornetTrapFeature>
)

data class HornetTrapFeature(
    val type: String,
    val geometry: HornetNestGeometry,
    val properties: HornetTrapProperties
)

data class HornetTrapProperties(
    @SerializedName("access_code") val accessCode: String,
    val name: String,
    @SerializedName("total_caught") val totalCaught: Int
)

File: ios/HivePulse/Network/HornetDTOs.swift
import Foundation

// MARK: - Stats

struct HornetStats: Codable {
    let totalCaught: Int
    let totalNests: Int
    let destroyedNests: Int
    let pendingSightings: Int
    let confirmedSightings: Int
    let totalTraps: Int

    init(totalCaught: Int, totalNests: Int, destroyedNests: Int,
         pendingSightings: Int, confirmedSightings: Int, totalTraps: Int = 0) {
        self.totalCaught = totalCaught
        self.totalNests = totalNests
        self.destroyedNests = destroyedNests
        self.pendingSightings = pendingSightings
        self.confirmedSightings = confirmedSightings
        self.totalTraps = totalTraps
    }

    enum CodingKeys: String, CodingKey {
        case totalCaught = "total_caught"
        case totalNests = "total_nests"
        case destroyedNests = "destroyed_nests"
        case pendingSightings = "pending_sightings"
        case confirmedSightings = "confirmed_sightings"
        case totalTraps = "total_traps"
    }
}

// MARK: - Catches

struct HornetCatchCreate: Codable {
    let latitude: Double?
    let longitude: Double?
    let count: Int
    let reporterName: String?

    enum CodingKeys: String, CodingKey {
        case latitude, longitude, count
        case reporterName = "reporter_name"
    }
}

struct HornetCatchOut: Codable {
    let id: String
    let count: Int
    let latitude: Double?
    let longitude: Double?
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, count, latitude, longitude
        case createdAt = "created_at"
    }
}

// MARK: - Nests

struct HornetNestCreate: Codable {
    let latitude: Double
    let longitude: Double
    let reporterName: String?
    let notes: String?
    let photoUrl: String?

    enum CodingKeys: String, CodingKey {
        case latitude, longitude, notes
        case reporterName = "reporter_name"
        case photoUrl = "photo_url"
    }
}

struct HornetNestOut: Codable, Identifiable {
    let id: String
    let latitude: Double
    let longitude: Double
    let status: String
    let reporterName: String?
    let notes: String?
    let photoUrl: String?
    let createdAt: Date
    let updatedAt: Date

    enum CodingKeys: String, CodingKey {
        case id, latitude, longitude, status, notes
        case reporterName = "reporter_name"
        case photoUrl = "photo_url"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

// MARK: - Nests GeoJSON

struct HornetNestGeoJSON: Codable {
    let type: String
    let features: [HornetNestFeature]
}

struct HornetNestFeature: Codable {
    let type: String
    let geometry: HornetNestGeometry
    let properties: HornetNestProperties
}

struct HornetNestGeometry: Codable {
    let type: String
    let coordinates: [Double] // [longitude, latitude]
}

struct HornetNestProperties: Codable, Identifiable {
    let id: String
    let status: String
    let reporterName: String?
    let notes: String?
    let photoUrl: String?
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id, status, notes
        case reporterName = "reporter_name"
        case photoUrl = "photo_url"
        case createdAt = "created_at"
    }
}

// MARK: - Sightings

struct HornetSightingCreate: Codable {
    let photoUrl: String
    let description: String?
    let reporterName: String?
    let latitude: Double?
    let longitude: Double?

    enum CodingKeys: String, CodingKey {
        case description, latitude, longitude
        case photoUrl = "photo_url"
        case reporterName = "reporter_name"
    }
}

struct HornetSightingOut: Codable, Identifiable {
    let id: String
    let photoUrl: String
    let description: String?
    let reporterName: String?
    let status: String
    let yesVotes: Int
    let noVotes: Int
    let createdAt: Date
    let latitude: Double?
    let longitude: Double?

    enum CodingKeys: String, CodingKey {
        case id, description, status, latitude, longitude
        case photoUrl = "photo_url"
        case reporterName = "reporter_name"
        case yesVotes = "yes_votes"
        case noVotes = "no_votes"
        case createdAt = "created_at"
    }
}

struct HornetVote: Codable {
    let vote: String
}

// MARK: - Traps

struct HornetTrapCreate: Codable {
    let name: String
    let latitude: Double
    let longitude: Double
    let notes: String?
    let ownerName: String?

    enum CodingKeys: String, CodingKey {
        case name, latitude, longitude, notes
        case ownerName = "owner_name"
    }
}

struct HornetTrapCatchCreate: Codable {
    let count: Int
    let caughtOn: String  // ISO date string YYYY-MM-DD

    enum CodingKeys: String, CodingKey {
        case count
        case caughtOn = "caught_on"
    }
}

struct HornetTrapCatchOut: Codable, Identifiable {
    let id: String
    let trapId: String
    let count: Int
    let caughtOn: String
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, count
        case trapId = "trap_id"
        case caughtOn = "caught_on"
        case createdAt = "created_at"
    }
}

struct HornetTrapOut: Codable, Identifiable {
    let id: String
    let accessCode: String
    let name: String
    let latitude: Double
    let longitude: Double
    let notes: String?
    let ownerName: String?
    let createdAt: Date
    let totalCaught: Int
    let catches: [HornetTrapCatchOut]

    enum CodingKeys: String, CodingKey {
        case id, name, latitude, longitude, notes, catches
        case accessCode = "access_code"
        case ownerName = "owner_name"
        case createdAt = "created_at"
        case totalCaught = "total_caught"
    }
}

struct HornetTrapNearbyOut: Codable {
    let accessCode: String
    let name: String
    let latitude: Double
    let longitude: Double
    let distanceM: Int
    let totalCaught: Int

    enum CodingKeys: String, CodingKey {
        case name, latitude, longitude
        case accessCode = "access_code"
        case distanceM = "distance_m"
        case totalCaught = "total_caught"
    }
}

struct HornetTrapsGeoJSON: Codable {
    let type: String
    let features: [HornetTrapFeature]
}

struct HornetTrapFeature: Codable {
    let type: String
    let geometry: HornetNestGeometry      // same shape: Point + [lon, lat]
    let properties: HornetTrapProperties
}

struct HornetTrapProperties: Codable {
    let accessCode: String
    let name: String
    let totalCaught: Int

    enum CodingKeys: String, CodingKey {
        case name
        case accessCode = "access_code"
        case totalCaught = "total_caught"
    }
}
